packages/cozy-sharing/src/SharingBanner/test/RouterContext.jsx

```diff
@@ -1,17 +1,18 @@
 import React, { useContext } from 'react'
-import { withRouter } from 'react-router'
+import { useLocation, useParams } from 'react-router-dom'
 
 export const RouterContext = React.createContext()
 
-export const RouterContextProvider = withRouter(
-  ({ children, router, params, location, routes }) => {
-    return (
-      <RouterContext.Provider value={{ router, params, location, routes }}>
-        {children}
-      </RouterContext.Provider>
-    )
-  }
-)
+export const RouterContextProvider = ({ children }) => {
+  const location = useLocation()
+  const params = useParams()
+
+  return (
+    <RouterContext.Provider value={{ params, location }}>
+      {children}
+    </RouterContext.Provider>
+  )
+}
```

Comment:
Can you explain within the commit message, why? 

withRouter seems to exist: https://v5.reactrouter.com/core/api/withRouter ? 

Also, I don't see react-router to be part as a dev-dep & peerDep https://github.com/cozy/cozy-libs/blob/master/packages/cozy-sharing/package.json 

So I think we have this regression here because one package has removed it's dep to react-router 

Also, some app consuming cozy-sharings have old react-router version: https://github.com/cozy/cozy-drive/blob/master/package.json#L164

Is it still working? After a quick search, I tend to think that no... useParams is not present here https://github.com/remix-run/react-router/blob/v3/docs/API.md

Reply:
Je pense que ce fix n'est pas bon en effet.

>So I think we have this regression here because one package has removed it's dep to react-router

Il faudrait clairement ajouter `react-router` aux devDeps/peerDeps et annuler ce changement 👍